common-arcana.lic

```diff
@@ -72,35 +72,42 @@ module DRCA
     end
   end
 
-  def start_barb_abilities(skills, settings)
-    skills
+  def start_barb_abilities(abilities, settings)
+    # Note, you must know Power meditation or Powermonger mastery
+    # for your active abilities to be detected by DRSpells.
+    abilities
       .reject { |name| DRSpells.active_spells[name] }
-      .each do |name|
-        activate_barb_buff?(name)
-        waitrt?
-        pause settings.meditation_pause_timer.to_i if get_data('spells').barb_abilities[name]['type'].eql?('meditation')
-        DRC.fix_standing
-      end
+      .each { |name| activate_barb_buff?(name, settings) }
+    DRC.fix_standing
   end
 
-  def activate_barb_buff?(name)
+  def activate_barb_buff?(name, settings = nil)
     ability_data = get_data('spells').barb_abilities[name]
     case DRC.bput(ability_data['start_command'], ability_data['activated_message'], 'But you are already', 'Your inner fire lacks', 'find yourself lacking the inner fire', 'You should stand', 'You must be sitting', 'You must be unengaged', 'While swimming?')
     when 'You must be unengaged'
       DRC.retreat
-      activate_barb_buff?(name)
+      activate_barb_buff?(name, settings)
     when 'You must be sitting'
       DRC.retreat
       case DRC.bput('sit', 'You sit', 'You are already', 'You rise', 'While swimming?')
       when 'While swimming?'
         false # can't sit here, water is too deep
       else
-        activate_barb_buff?(name)
+        activate_barb_buff?(name, settings)
       end
     when 'You should stand'
       DRC.fix_standing
-      activate_barb_buff?(name)
-    when ability_data['activated_message']
+      activate_barb_buff?(name, settings)
+    when /#{ability_data['activated_message']}/
```

Comment:
Express the match statement as regular expression because in `base-spells.yaml` the activated message is `^You .* to meditate`. The bug was that the case statement was trying to match on that literally. Now it'll match as a regex.

Reply:
Nice catch!